week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,27 +1,118 @@
-function setAlarm() {}
+let timeLeft;
+let intervalId;
+let flashingIntervalId;
+let isPaused = false;
+let isRedBackground = false;
+let isBlueBackground = true;
 
-// DO NOT EDIT BELOW HERE
+function setAlarm() {
+  reset();
 
-function setup() {
-  var audio = new Audio("alarmsound.mp3");
+  timeLeft = Number(document.getElementById("alarmSet").value);
```

Comment:
It seems `timeLeft` is assigned a value here but then the rest of the code has to ensure it's valid.

I'd suggest to validate it before doing anything else to simplify the code.